go/tools/builders/env.go

```diff
@@ -332,6 +336,40 @@ func absArgs(args []string, flags []string) {
 	}
 }
 
+
+// absToRelativePathWriter filters the output stream from a compile command,
+// replacing any absolute paths with relative equivalents.
+//
+// This is done on a best-effort basis, and not much effort is undertaken. If a
+// path is split across multiple Write calls, it will not be relativized.
```

Comment:
I don't think this is good behavior. It would be better to buffer all the output, then replace paths all at once.

Reply:
That works. In that case, I can consolidate the implementation into just the relativizePaths func.